### PR TITLE
[Woo POS] Update products list UI 

### PR DESCRIPTION
**Pull request by bozidarsevo**
Closes: #13414

## Description

Updated main parts of the Products list view
- updated info banner view
- updated icons
- updates fonts and colors
- added shadow
- making content full width and height and make background colors go behind safe area

**Disclaimer**: some shadows, paddings and similar things might need tweaks later on when we update other parts of the POS UI

## Testing information

- Enter POS
- Make sure the products are shown
- Close the top info banner
- Tap few products and check that they are added to the cart

## Screenshots

| Before      | After |
| ----------- | ----------- |
|   Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 55 57    | Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 50 55       |
| Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 44 02      | Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 50 58       |
| Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 44 10     | Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 51 03       |

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/ItemCardView.swift [hunk 1]
```diff
@@ -16,7 +16,6 @@ struct ItemCardView: View {
                 ProductImageThumbnail(productImageURL: URL(string: imageSource),
                                       productImageSize: Constants.productCardHeight,
                                       scale: scale,
-                                      productImageCornerRadius: Constants.productImageCornerRadius,
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Presentation/ItemCardView.swift:19]: We do not need corner radius on the product image, we have one for the whole view

**Review by staskus — COMMENTED** [review]
Observation (probably for another pr) that now when layout goes below safe area we could also move the floating button down:

image

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift [hunk 2]
```diff
@@ -186,7 +214,7 @@ private extension ItemListView {
         )
         static let headerBannerHint = NSLocalizedString(
             "pos.itemlistview.headerBannerHint",
-            value: "Other product types, such as variable and virtual, will become available in future updates.",
+            value: "Other product types, such as variable and virtual, will become available in future updates. Learn more",
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/ItemListView.swift:217]: Do we know what we want to open with this "Learn more"? I think we can skip this change until we implement that.

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift [hunk 3]
```diff
@@ -159,13 +182,18 @@ private extension ItemListView {
 private extension ItemListView {
     enum Constants {
         static let titleFont: Font = .system(size: 40, weight: .bold, design: .default)
-        static let bannerTitleFont: Font = .system(size: 26, weight: .bold, design: .default)
-        static let bannerHeight: CGFloat = 120
-        static let infoIconSize: CGFloat = 48
+        static let bannerTitleFont: Font = .system(size: 24, weight: .bold, design: .default)
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/ItemListView.swift:185]: My observation with both PRs is that designs use some different fonts than the default Apple ones. Not sure if you observe the same. Of course, if needed we could make some replacements later.

**Review by staskus — APPROVED** [review]
👍

**Comment by bozidarsevo** [comment]
> Observation (probably for another pr) that now when layout goes below safe area we could also move the floating button down:
> 
> image

Yep! I noticed that also and will move it a bit down, tnx!

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift (hunk 2 above)
Comment (bozidarsevo) [reply WooCommerce/Classes/POS/Presentation/ItemListView.swift:217]: I will check if that also opens the full banner or something else!

**Pull request merged**